### PR TITLE
extending Volume button feature, by scroll amount

### PR DESCRIPTION
**Pull request by r1di**
Added volume button scroll amount

### CODE REVIEW

File: android/app/src/main/java/com/rajarsheechatterjee/VolumeButtonListener/VolumeButtonListener.java [hunk 1]
```diff
@@ -27,22 +27,33 @@ public static boolean isActive() {
...
+    private static int scrollAmount = 100;
+
+    public static void setScrollAmount(int amount) {
+        scrollAmount = amount;
     }
 
-    public static void up() {
-        sendEvent(true);
+    public static int getScrollAmount() {
+        return scrollAmount;
```
Comment (nyagami) [inline android/app/src/main/java/com/rajarsheechatterjee/VolumeButtonListener/VolumeButtonListener.java:56]: Why we have to send event with amount value

Comment (r1di) [reply android/app/src/main/java/com/rajarsheechatterjee/VolumeButtonListener/VolumeButtonListener.java:56]: the parameter "amount" is the value representing the amount of scrolling desired when either the volume up or volume down button is pressed.

in the MainActivity class, the dispatchKeyEvent() method listens for volume button presses (up and down) and calls the respective VolumeButtonListener.up() and VolumeButtonListener.down() methods with the scroll amount obtained from VolumeButtonListener.getScrollAmount()

Comment (nyagami) [reply android/app/src/main/java/com/rajarsheechatterjee/VolumeButtonListener/VolumeButtonListener.java:56]: Actually the value of `scrollAmout` is stored and persisted by `Redux` (when you call dispatch to settingReducer).
You could remove old eventListener and add the new one when `scrollAmout` get changed.
No need to modify the naitve Module. Anyway I think your logic is overhandling
image

Comment (r1di) [reply android/app/src/main/java/com/rajarsheechatterjee/VolumeButtonListener/VolumeButtonListener.java:56]: Thanks, I will look into it. 
Never worked before with Redux/JavaScript, gotta use more chadGPT for knowledge gaps😂👌

File: src/screens/settings/SettingsReaderScreen/SettingsReaderScreen.tsx [hunk 2]
```diff
@@ -196,6 +195,28 @@ const SettingsReaderScreen = () => {
...
           theme={theme}
         />
+        {useVolumeButtons ? (
+          <View style={styles.scrollAmount}>
+            <Text style={[labelStyle, styles.paddingRightM]} numberOfLines={2}>
+              {getString('readerScreen.bottomSheet.scrollAmount')}
+            </Text>
+            <TextInput
+              style={{
+                paddingHorizontal: 16,
```
Comment (r1di) [inline src/screens/settings/SettingsReaderScreen/SettingsReaderScreen.tsx:205]: could have used a StyleSheet i guess?

File: android/app/src/main/java/com/rajarsheechatterjee/VolumeButtonListener/VolumeButtonListener.java (hunk 1 above)
Comment (r1di) [reply android/app/src/main/java/com/rajarsheechatterjee/VolumeButtonListener/VolumeButtonListener.java:56]: i pushed a new commit, not sure if this was the correct way(im coming from gerrit, is this like a new patchset?).
I think its cleaner now, now working with redux for all the changes.

I tried to use **KeyboardAvoidingView**, but didnt manage to implement it, would be usefull for the future.

thank you for patience😁

**Review by rajarsheechatterjee — APPROVED** [review]

File: android/app/src/main/java/com/rajarsheechatterjee/VolumeButtonListener/VolumeButtonListener.java (hunk 1 above)
Comment (SkillGG) [reply android/app/src/main/java/com/rajarsheechatterjee/VolumeButtonListener/VolumeButtonListener.java:56]: The native module is ONLY to notify ReactNative that the button has been pressed. The rest of the behaviour SHOULD BE IN REACT NATIVE not in java...

**Pull request merged**